src/account/account.cairo

```diff
@@ -71,9 +70,9 @@ mod AccountComponent {
 
             // Check tx version
             let tx_info = get_tx_info().unbox();
-            let version = tx_info.version;
-            if version != TRANSACTION_VERSION {
-                assert(version == QUERY_VERSION, Errors::INVALID_TX_VERSION);
+            if (tx_info.version != QUERY_VERSION) {
+                let tx_version: u64 = tx_info.version.try_into().unwrap();
+                assert(MIN_TRANSACTION_VERSION <= tx_version, Errors::INVALID_TX_VERSION);
             }
```

Comment:
```suggestion
            let tx_version: u64 = tx_info.version.try_into().unwrap();
            assert(MIN_TRANSACTION_VERSION <= tx_version, Errors::INVALID_TX_VERSION);
```
i don't think this `if` serves any purpose now

Reply:
The suggestion was the original idea with casting to u256. We need the `if` when we cast to smaller uints because `QUERY_VERSION` is too big (2**128 + tx version)

Reply:
is adding the `if` cheaper than casting to a smaller uint? the issue with the current version is that it doesn't support v3 or higher simulations, only v2. we'd need `<=`. would that also be cheaper?

Reply:
Mmmm right. That case, I think u256 is the way to go to cover all cases because we can't use `<=` without casting and simulations would need to be u256 anyway. All the efficiency work is for naught. I'll add a test for higher simulations